lib/shopify_api/event_pipe/event_queue.ex

```diff
@@ -61,31 +59,23 @@ defmodule ShopifyAPI.EventPipe.EventQueue do
 
   defp enqueue_event(worker, event, opts) do
     Logger.warn(fn -> "Enqueueing in default queue #{inspect(event)}" end)
-    background_job_impl().(Exq, "default", worker, [event], opts)
+    background_job_enqueue("default", worker, [event], opts)
     {:ok}
   end
 
-  def register(token) do
-    case GenServer.whereis(Exq) do
-      nil ->
-        :timer.sleep(@retry_timer)
-        register(token)
+  def subscribe(token) do
+    background_job_impl().subscribe(token)
+  end
 
-      _res ->
-        Logger.info(fn -> "#{__MODULE__} registering #{AuthToken.create_key(token)}" end)
-        Exq.subscribe(Exq, AuthToken.create_key(token), 1)
-    end
+  defp background_job_enqueue(queue_name, worker, events, opts) do
```

Comment:
is this method even needed any more?

Reply:
I am on the fence on this one. It is only called twice and so it is not really adding any value here.

Reply:
Mostly recommend removing it as it doesn't hide much complexity and would actually simplify the code base there.